### PR TITLE
417 upload incident

### LINKED ISSUE #223: Officer in action section: Show lawsuits, media articles
It would be great if we could get videos of encounters with this officer that we get through FOIA, SecureDrop, or otherwise associated with the officer and shown on their profile.

### LINKED ISSUE #417: Add incident upload for area coordinators
We'd want to store only the public data:

* Date/time of incident (rounded to hour)
* Exact location (rounded to block)
* Incident report # (this is something that would be specific to the local organization, so they can check against their private report to look up e.g. witnesses of a specific incident) 
* Officers involved (link to officer ID) 
* License numbers
* Description of incident
* We do _not_ want to store the victim or witness information.

*Editable by:* Administrators, area coordinates (requires #355)

Copwatch Berkeley incident form for reference:

screen shot 2018-04-11 at 9 19 37 pm

### LINKED ISSUE #418: Add video upload for area coordinators
We're not going to store video, but we can link out to it on e.g. Youtube. We should provide a video upload for area coordinators, so they can submit a video and denote which incident (requires implementation of #417) it corresponds to.

### PR DESCRIPTION
**Pull request by raq929**
## Status

Ready for re-review

- Also @redshiftzero can you let me know how you want users to get to the incidents list? Should we just link it in the nav?
- Needs review of mobile styles. (Incidents seems ok, but on the officers page the same html doesn't work the same way. 😠) I suspect there's a missing end tag somewhere on the officer page.

## Description of Changes
There are some big changes here, including changes to the file structure for route testing.
There are now incidents, which have relationships with locations, as well as license plates, links, and officers. Officers should be able to list their incidents and vice versa.
This PR also creates a ModelView, based on the Flask MethodView and Django's ModelViews, which is a reusable API that should let us get new models out the door quickly. It's being used for incidents right now. (I initially thought I would be making separate views for Links and LicensePlates, but later encountered FormFields and didn't go that route.)

For the future:
Create views for admins to edit/delete Locations, LicensePlates, and Links. Because they have many to many relationships with Incidents, they are not deleted when Incidents are deleted, so it would be convenient to have a way to delete ones that are hanging around without an incident. This should hopefully be a simple matter of plugging in the appropriate variables into the ModelView and creating some templates (this PR also creates some generic form and list templates to inherit from.)

Fixes #417
Fixes #418 
Fixes #223.

Changes proposed in this pull request:

 - Adds incidents, including Locations, Links, LicensePlates
 - Creates views and templates for incidents
 - Adds links to officers and updates the appropriate forms and templates

## Screenshots (if appropriate)
screen shot 2018-05-11 at 5 44 16 pm

## Tests and linting
 
 [ x ] I have rebased my changes on current `develop`
 
 [ x ] pytests pass in the development environment on my local machine
One of the browser tests timed out. I'm hoping it passes CI
 
 [ x ] `flake8` checks pass

### CODE REVIEW

File: OpenOversight/app/templates/incident_new.html [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{% extends 'form.html' %}
```
Comment (redshiftzero) [inline OpenOversight/app/templates/incident_new.html]: On the `/volunteer` page we have these buttons at the bottom - what about we move them to the top, and add a button for "New Incident":

screen shot 2018-05-14 at 9 04 55 pm

File: OpenOversight/app/templates/partials/incident_form.html [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+			{% endwith %}
+			<div>
+				<legend>{{ form.license_plates.label }}</legend>
+				{% for subform in form.license_plates %}
+					{% with id="js-license-plate", number=loop.index %}
+						{% include "partials/subform.html" %}
+					{% endwith %}
+				{% endfor %}
+				{# buttons are disabled until the DOM loads and click handlers are added #}
+				<button class="btn btn-success js-add-another-button" disabled>Add another license plate</button>
```
Comment (redshiftzero) [inline OpenOversight/app/templates/partials/incident_form.html]: Small bug: If click remove when there is a single license plate, then clicking "Add another license plate" doesn't do anything

File: OpenOversight/app/templates/partials/links_subform.html [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<div>
+	<legend>{{form.links.label}}</legend>
+	{% for subform in form.links %}
+		{% include "partials/subform.html" %}
+	{% endfor %}
+	<button class="btn btn-success js-add-another-button" disabled>Add another link</button>
```
Comment (redshiftzero) [inline OpenOversight/app/templates/partials/links_subform.html]: Similarly here if I click "Remove" if there is only one link, then "Add another link" doesn't do anything

File: OpenOversight/app/templates/partials/links_and_videos_row.html [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+		{% endif %}
+		{% if type == 'video' %}
+			<div class="col-sm-12 col-md-6">
+				<ul class='list-group'>
+					<h5>Videos</h5>
+					{% for link in list %}
+						{% with link_url = link.url.split('v=')[1] %}
+							<li class='list-group-item'>
+								<div class='video-container'>
+									<iframe width="560" height="315" src="https://www.youtube.com/embed/{{ link_url }}" frameborder="0" allow="autoplay; encrypted-media" allowfullscreen></iframe>
```
Comment (redshiftzero) [inline OpenOversight/app/templates/partials/links_and_videos_row.html]: I do think we want to enable links to other videos - but they don't necessarily need to be embedded in non-Youtube cases - we just want to have "Videos" in a separate area together on the incident page (away from other link types, e.g. media articles)

File: OpenOversight/app/templates/officer.html [hunk 5]
```diff
@@ -58,112 +58,127 @@ <h3>General Information
...
-    </div>
-  </div>
-</div>
+          </form>
+        </tbody>
+      </table>
+      {% endif %}
+    </div> {# end col #}
+    <div class='col-sm-12 col-md-6'>
+      <h3>Incidents</h3>
```
Comment (redshiftzero) [inline OpenOversight/app/templates/officer.html]: This profile page is looking sweet 😎 . We should also have a link to the edit incident page like we do on the general information page next to the incident:

screen shot 2018-05-14 at 9 33 38 pm

And an "Add incident" button that takes us to the new incident page: 

screen shot 2018-05-14 at 9 34 17 pm

File: OpenOversight/app/templates/partials/incident_form.html (hunk 2 above)
Comment (raq929) [reply OpenOversight/app/templates/partials/incident_form.html]: OH YEAH, I meant to fix that. 😛

File: OpenOversight/app/templates/partials/links_and_videos_row.html (hunk 4 above)
Comment (raq929) [reply OpenOversight/app/templates/partials/links_and_videos_row.html]: Cool. I'll add a 'non YouTube video' link type, that way they can be with the videos but not embedded.

**Pull request merged**